middleware/auth.go

```diff
@@ -4,18 +4,24 @@ import (
 	"net/http"
 	"pay/controllers"
 	_ "pay/controllers"
-	"pay/models"
+	"pay/core"
 
 	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
 )
 
 var UnauthenticatedError = gin.H{"error": "unauthenticated"}
 
 func Auth(c controllers.Controller) gin.HandlerFunc {
 	return func(ctx *gin.Context) {
-		u := models.User{}
-		uuid := ctx.Param("user_uuid")
-		err := c.DB.Model(models.User{}).Where("UUID = ?", uuid).Take(&u).Error
+		UUIDstr := ctx.Param("user_uuid")
+		UUID, err := uuid.FromBytes([]byte(UUIDstr))
```

Comment:
Щодо твоєї помилки з `/hello`, то причина в цьому. Тут треба юзати [`uuid.Parse`](https://pkg.go.dev/github.com/google/uuid#Parse), бо ми читаємо текстовий формат, а не голі байти.

```suggestion
		UUID, err := uuid.Parse(UUIDstr)
```